include/boost/spirit/home/qi/numeric/detail/numeric_utils.hpp

```diff
@@ -316,7 +316,7 @@ namespace boost { namespace spirit { namespace qi { namespace detail
             if (!Accumulate)
             {
                 // skip leading zeros
-                while (it != last && *it == '0' && leading_zeros < MaxDigits)
+                while (it != last && *it == '0' && (MaxDigits < 0 || leading_zeros < static_cast< std::size_t >(MaxDigits)))
```

Comment:
On Tue, Jul 14, 2015 at 12:50 PM, Joel de Guzman <notifications@github.com>
wrote:

> In include/boost/spirit/home/qi/numeric/detail/numeric_utils.hpp
> <https://github.com/boostorg/spirit/pull/140#discussion_r34552130>:
>
> > @@ -316,7 +316,7 @@ namespace boost { namespace spirit { namespace qi { namespace detail
> >              if (!Accumulate)
> >              {
> >                  // skip leading zeros
> > -                while (it != last && *it == '0' && leading_zeros < MaxDigits)
> > +                while (it != last && *it == '0' && (MaxDigits < 0 || leading_zeros < static_cast< std::size_t >(MaxDigits)))
>
> Why did you add the logic with MaxDigits < 0 ? I don' think this will ever
> happen in this code branch.
>
1. I didn't find any restrictions on the MaxDigits values. From the
specialization below it seems MaxDigits == -1 is allowed and means "no
limit". I don't know what behavior should be for other negative values, so
I extended the same semantics on them as well. Let me know if this
assumption is wrong.
2. I found only one specialization of extract_int, which is the one below -
for MinDigits == 1 and MaxDigits == -1. This specialization won't be used
for e.g. MinDigits == 3 && MaxDigits == -1 which means the generic template
will be used and the 'while' condition should be prepared for negative
values.